property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
+        break;
+      case 'date':
+        console.warn('Warning: property default value was converted' +
+          ' from string using ISO formatting');
+        if (value.toLowerCase() === 'now'){
+          def.defaultFn = 'now';
+        } else {
+          def.default = new Date(value);
+        }
+        break;
+      case 'geopoint':
+        if (value.indexOf('lat') !== -1 && value.indexOf('lng') !== -1) {
+          def.default = JSON.parse(value);
+        } else {
+          var geo = value.replace(/[\s,]+/g, ',').split(',');
+          def.default = {};
+          def.default.lat = Number(geo[0]);
+          def.default.lng = Number(geo[1]);
+        }
+        break;
+      case 'buffer':
+        console.warn('Warning: property default value was converted' + 
+          'from string using UTF8 encoding');
```

Comment:
This will make it difficult for people to provide binary data. Can we use Base64 encoding instead please? AFAIK, `Buffer` constructor supports that too.